### PR TITLE
adds pause to action menu

### LINKED ISSUE #931: Add "Pause" option in the actions menu on the dashboard
Currently, in order to pause a monitor, one should click on the monitor and then the Configure button. Adding this option to the action menu will speed it up.

Screenshot 2024-10-11 at 12 35 25 AM

### PR DESCRIPTION
**Pull request by Sinon11901**
This change resolves #931 by adding pause option in the actions menu on the dashboard.

### CODE REVIEW

File: Client/src/Pages/Monitors/Home/actionsMenu.jsx [hunk 1]
```diff
@@ -44,6 +47,28 @@ const ActionsMenu = ({ monitor, isAdmin, updateCallback }) => {
...
+      if (pauseUptimeMonitor.fulfilled.match(action)) {
+        updateCallback();
+        //window.location.reload();
+        if (action.payload?.isActive === false){
+          createToast({ body: "Monitor paused successfully." });
+        }else{
+          createToast({ body: "Monitor resumed successfully." });
+        }
+      } else {
+        throw new Error(action.error.message);
```
Comment (ajhollid) [inline Client/src/Pages/Monitors/Home/actionsMenu.jsx:64]: It's a good idea to safely access nested properties:

```
  throw new Error(action?.error?.message ?? "Failed to pause monitor.");
```

`action` isn't guaranteed to have `error` defined, so we shoudl fail gracefully if it isn't

File: Client/src/Pages/Monitors/Home/actionsMenu.jsx [hunk 2]
```diff
@@ -44,6 +47,28 @@ const ActionsMenu = ({ monitor, isAdmin, updateCallback }) => {
...
+      }
+    } catch (error) {
+      console.error("Error pausing monitor:", monitor._id, error);
```
Comment (ajhollid) [inline Client/src/Pages/Monitors/Home/actionsMenu.jsx:67]: We have a utility Logger that we should use for logging:

```
  import { logger } from "../../../Utils/Logger";
  ...
  logger.error("Error pausing monitor:", monitor._id, error);
```

The logger was added mid-projecyt, so if you see `console.log` statements throughout the app we just haven't gotten around to replacing them yet.

File: Client/src/Pages/Monitors/Home/actionsMenu.jsx [hunk 3]
```diff
@@ -155,6 +180,20 @@ const ActionsMenu = ({ monitor, isAdmin, updateCallback }) => {
...
+            onClick={(e) => {
+              e.stopPropagation();
+              handlePause(e);
+            }}
+          >
+          {monitor?.isActive ? (
+            "Pause"
+          ) : (
+            "Resume"
+          )}
```
Comment (ajhollid) [inline Client/src/Pages/Monitors/Home/actionsMenu.jsx:194]: The regular braces are not needed here:

`{monitor?.isActive  === true ? "Pause" : "Resume"}`

is sufficient :+1:

File: Client/src/Pages/Monitors/Home/actionsMenu.jsx [hunk 4]
```diff
@@ -44,6 +47,28 @@ const ActionsMenu = ({ monitor, isAdmin, updateCallback }) => {
     }
   };
 
+  const handlePause = async (event) => {
```
Comment (ajhollid) [inline Client/src/Pages/Monitors/Home/actionsMenu.jsx:50]: `event` parameter is unused, we can safely remove it

File: Client/src/Pages/Monitors/Home/actionsMenu.jsx [hunk 5]
```diff
@@ -44,6 +47,28 @@ const ActionsMenu = ({ monitor, isAdmin, updateCallback }) => {
...
+    try {
+      const action = await dispatch(
+        pauseUptimeMonitor({ authToken, monitorId: monitor._id })
+      );
+      if (pauseUptimeMonitor.fulfilled.match(action)) {
+        updateCallback();
+        //window.location.reload();
```
Comment (ajhollid) [inline Client/src/Pages/Monitors/Home/actionsMenu.jsx:57]: We can remove this commented out code too

**Review by ajhollid — CHANGES_REQUESTED** [review]
Looks great for the most part!  A couple of minor changes that I think clean things up a bit :+1:  Thanks for your hard work here!

**Review by ajhollid — APPROVED** [review]
Looks good to me, thanks for making those changes :+1:

**Comment by Sinon11901** [comment]
Thank you for your cooperation and quick feedback. I really enjoyed working on this project.

**Pull request merged**